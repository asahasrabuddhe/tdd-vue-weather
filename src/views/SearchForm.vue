<template>
    <div class="level-item">
        <form @submit.prevent="$emit('submit', city)" class="field has-addons">
            <p class="control"><input class="input" type="text" v-model="city" placeholder="Enter City" /></p>
            <p class="control"><button class="button" type="submit">Find Weather</button></p>
        </form>
    </div>
</template>

<script>
    export default {
        name: "SearchForm",
        data() {
            return {
                city: ''
            }
        }
    }
</script>

<style scoped>

</style>
