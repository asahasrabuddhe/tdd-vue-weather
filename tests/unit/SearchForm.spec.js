import SearchForm from "@/views/SearchForm";
import {shallowMount} from "@vue/test-utils";

describe('SearchForm', () => {
    const build = () => {
        const wrapper = shallowMount(SearchForm)

        return {
            wrapper,
            input: () => wrapper.find('input'),
            button: () => wrapper.find('button')
        }
    }

    it('renders the component properly', () => {
        const {wrapper} = build()

        expect(wrapper.html()).toMatchSnapshot()
    })

    it('renders its contents properly', () => {
        const {input, button} = build()

        expect(input().exists()).toBeTruthy()
        expect(button().exists()).toBeTruthy()
    })

    it('calls the submit event when the form is submitted', () => {
        const expectedCity = 'Pune'
        const {wrapper, button, input} = build()

        input().element.value = expectedCity

        input().trigger('input')
        button().trigger('click')
        button().trigger('submit')

        expect(wrapper.emitted().submit[0]).toEqual([expectedCity])
    })
})
