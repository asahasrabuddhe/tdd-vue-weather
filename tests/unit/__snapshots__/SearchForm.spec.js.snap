// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders the component properly 1`] = `
<div class="level-item">
  <form class="field has-addons">
    <p class="control"><input type="text" placeholder="Enter City" class="input"></p>
    <p class="control"><button type="submit" class="button">Find Weather</button></p>
  </form>
</div>
`;
